presto-cache/src/main/java/com/facebook/presto/cache/CacheConfig.java

```diff
@@ -73,31 +68,41 @@ public CacheConfig setMaxInMemoryCacheSize(DataSize maxInMemoryCacheSize)
         return this;
     }
 
-    @Min(1)
-    public int getMaxCachedEntries()
+    @Config("cache.enabled")
+    @ConfigDescription("Is cache enabled")
+    public CacheConfig setCachingEnabled(boolean cachingEnabled)
+    {
+        this.cachingEnabled = cachingEnabled;
+        return this;
+    }
+
+    public boolean isCachingEnabled()
     {
-        return maxCachedEntries;
+        return cachingEnabled;
     }
 
-    @Config("cache.max-cached-entries")
-    @ConfigDescription("Number of entries allowed in the cache")
-    public CacheConfig setMaxCachedEntries(int maxCachedEntries)
+    @Config("cache.type")
+    @ConfigDescription("Caching type")
+    public CacheConfig setCacheType(CacheType cacheType)
     {
-        this.maxCachedEntries = maxCachedEntries;
+        this.cacheType = cacheType;
         return this;
     }
 
-    @MinDuration("0s")
-    public Duration getCacheTtl()
+    public CacheType getCacheType()
     {
-        return cacheTtl;
+        return cacheType;
     }
 
-    @Config("cache.ttl")
-    @ConfigDescription("Time-to-live for a cache entry")
-    public CacheConfig setCacheTtl(Duration cacheTtl)
+    @Override
+    public String toString()
     {
-        this.cacheTtl = cacheTtl;
-        return this;
+        return "CacheConfig{" +
```

Comment:
Can we use toStringHelper for this method?

Reply:
Done.